### PR TITLE
Move docs/sources/ to docs/sources/mimir/

### PR DESCRIPTION
**Pull request by andyasp**
#### What this PR does
Moves the contents of `docs/sources` to `docs/sources/mimir` to facilitate separating the helm docs.
The tricky part of this was fixing `embedmd` relative references, since they required another level of `../`.

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir/issues/3532

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -118,6 +118,7 @@
 
 ### Documentation
 
+* [CHANGE] Moved the contents of `docs/sources` to `docs/sources/mimir`. #3549
```
Comment (osg-grafana) [inline CHANGELOG.md:121]: Nice catch! :)

File: Makefile [hunk 2]
```diff
@@ -60,14 +60,14 @@ MIXIN_OUT_PATH_SUFFIXES := "" "-baremetal"
 JSONNET_MANIFESTS_PATH := operations/mimir
 
 # Doc templates in use
-DOC_TEMPLATES := docs/sources/operators-guide/configure/reference-configuration-parameters/index.template
+DOC_TEMPLATES := docs/sources/mimir/operators-guide/configure/reference-configuration-parameters/index.template
```
Comment (osg-grafana) [inline Makefile:63]: Got it! This is where I had problems. Thank you for the fix. :)

Comment (andyasp) [reply Makefile:63]: Happy to help!

File: Makefile [hunk 3]
```diff
@@ -60,14 +60,14 @@ MIXIN_OUT_PATH_SUFFIXES := "" "-baremetal"
...
-DOC_EMBED := docs/sources/operators-guide/configure/configure-the-query-frontend-work-with-prometheus.md \
-	docs/sources/operators-guide/configure/mirror-requests-to-a-second-cluster/index.md \
-	docs/sources/operators-guide/architecture/components/overrides-exporter.md \
-	docs/sources/operators-guide/get-started/_index.md \
-	docs/sources/operators-guide/deploy-grafana-mimir/jsonnet/deploy.md
+DOC_EMBED := docs/sources/mimir/operators-guide/configure/configure-the-query-frontend-work-with-prometheus.md \
+	docs/sources/mimir/operators-guide/configure/mirror-requests-to-a-second-cluster/index.md \
+	docs/sources/mimir/operators-guide/architecture/components/overrides-exporter.md \
+	docs/sources/mimir/operators-guide/get-started/_index.md \
+	docs/sources/mimir/operators-guide/deploy-grafana-mimir/jsonnet/deploy.md
```
Comment (osg-grafana) [inline Makefile:70]: I will look into this later to fully understand what is going on, but thank you for finding these embeds. (I am wondering why these would need to be done with the `Makefile` and not via a Hugo shortcode. Out-of-scope for this PR.)

**Comment by osg-grafana** [comment]
So far, so good! It is no wonder why my builds were not passing. Lots of locations to change! :)

**Review by osg-grafana — APPROVED** [review]
Unblocking for when all the builds pass on your time zone. :)

**Comment by osg-grafana** [comment]
Please coordinate with @replay if need be when merging this to `main`.

**Comment by andyasp** [comment]
I noticed more relative paths while double checking this. Converted to a draft while I sift through them.

**Comment by osg-grafana** [comment]
Looks like there are a bunch of images that need to be moved as well.

**Review by dimitarvdimitrov — APPROVED** [review]
SGTM, but I don't have experience with the docs, so maybe Jack is the right person to review this

File: docs/sources/mimir/tutorials/play-with-grafana-mimir/index.md [hunk 4]
```diff
@@ -47,10 +47,10 @@ In this tutorial, you'll:
    ```
 1. Navigate to the tutorial directory:
    ```bash
-   cd docs/sources/tutorials/play-with-grafana-mimir/
+   cd docs/sources/mimir/tutorials/play-with-grafana-mimir/
```
[the remark this answers is not in the document]
Comment (jdbaldry) [reply docs/sources/mimir/tutorials/play-with-grafana-mimir/index.md:50]: Good call out. I don't believe this should break. The website source file will still be the same and is used in the mount in https://github.com/grafana/website/blob/3480cee0d777d3d6c823c915f0f2ed9b1905d2b0/config/_default/config.yaml#L141-L142.

I've noticed that we hard code a version in there though so I have opened an issue to support the virtual latest in https://github.com/grafana/website/issues/10890

**Comment by jdbaldry** [comment]
I wonder if we might benefit from trying to reduce the number of of places this needs to be changed. Perhaps with a variable in the Makefile? This is probably the last time we move these files... until the next time :)

**Comment by replay** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (andyasp) [reply CHANGELOG.md]: Thanks, removed the entry.

**Comment by pracucci** [comment]
Could you rebase, please? If it's too complicated to pick the `.png` from `main` let me know and I can re-generate them here.

**Comment by andyasp** [comment]
> Looks like there are a bunch of images that need to be moved as well.

@osg-grafana Thanks, I'll rebase to fix these.

> I would have expected a change in `.github/workflows/publish-technical-documentation*` as well to update the source_folder.
> 
> And quick check that other workflows as well.

@krajorama Hidden folders strike again, nice catch! I updated these and searched for other uses.

> I wonder if we might benefit from trying to reduce the number of places this needs to be changed. Perhaps with a variable in the Makefile? This is probably the last time we move these files... until the next time 

@jdbaldry I was getting this same feeling too as these changes piled up. The different contexts the paths are used in (sh, go, yml, mk, md, are there more?!) scared me away from the idea.

**Review by pracucci — APPROVED** [review]
I took a quick look and I can't see anything wrong, so LGTM! Given the large changes (but low risk), I don't mind if we want to merge it quickly and then iterate on it if something is broken.

Ping me if you want to merge and you need someone to click the button. Thanks!

**Pull request merged**